src/libsyncengine/jobs/network/API_v2/upload_session/driveuploadsession.cpp

```diff
@@ -109,4 +111,27 @@ bool DriveUploadSession::handleCancelJobResult(const std::shared_ptr<UploadSessi
 
     return true;
 }
+
+void DriveUploadSession::abort() {
+    AbstractUploadSession::abort();
+    setVfsForceStatusCallback(nullptr);
+
+#ifdef __APPLE__
+    // Removing all extended attributes so that the file that failed to be uploaded is not identified as a dehydrated placeholder
+    // by the application when restarting the synchronization.
+
+    const SyncPath &localPath = getFilePath();
+    static const std::vector<const char *> extendedAttributes = {EXT_ATTR_STATUS, EXT_ATTR_PIN_STATE};
+
+    for (const auto attribute: extendedAttributes) {
```

Comment:
This should be in a utility function outside this class. Especially because `normal`upload (meaning direct uploads, not upload session) are also concerned.